nim_status/account.nim

```diff
@@ -56,16 +78,7 @@ proc child(self: ExtendedPrivKey, child: PathLevel): ExtendedPrivKeyResult =
   err($sk.error())
 
 proc derive*(seed: Keyseed, path: KeyPath): SecretKeyResult =
-  let hmacResult = sha512.hmac("Bitcoin seed", seq[byte] seed)
-  let secretKey = hmacResult.data[0..31]
-  let chainCode = hmacResult.data[32..63]
-  let sk = SkSecretKey.fromRaw(secretKey)
-  if sk.isErr(): return err("Invalid secret key")
-
-  var extPrivK = ExtendedPrivKey(
-    secretKey: sk.get(),
-    chainCode: chainCode
-  )
+  var extPrivK = splitHMAC(cast[string](seed), masterSecret).get()
```

Comment:
casting to string is not safe, the in-memory representation is not the same - instead, `string.fromBytes` should be used throughout (`stew/byteutils`)

Reply:
Thanks, will change it!